[2024-11-24T15:18:54.836+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-11-24T15:18:54.931+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: fraud_detection.process_transaction scheduled__2024-11-24T15:17:00+00:00 [queued]>
[2024-11-24T15:18:54.978+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: fraud_detection.process_transaction scheduled__2024-11-24T15:17:00+00:00 [queued]>
[2024-11-24T15:18:54.979+0000] {taskinstance.py:2865} INFO - Starting attempt 1 of 2
[2024-11-24T15:18:55.065+0000] {taskinstance.py:2888} INFO - Executing <Task(BranchPythonOperator): process_transaction> on 2024-11-24 15:17:00+00:00
[2024-11-24T15:18:55.105+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=927) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-11-24T15:18:55.113+0000] {standard_task_runner.py:72} INFO - Started process 932 to run task
[2024-11-24T15:18:55.108+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'fraud_detection', 'process_transaction', 'scheduled__2024-11-24T15:17:00+00:00', '--job-id', '594', '--raw', '--subdir', 'DAGS_FOLDER/fraud_detection_dag.py', '--cfg-path', '/tmp/tmpw7hxe0k8']
[2024-11-24T15:18:55.133+0000] {standard_task_runner.py:105} INFO - Job 594: Subtask process_transaction
[2024-11-24T15:18:55.306+0000] {task_command.py:467} INFO - Running <TaskInstance: fraud_detection.process_transaction scheduled__2024-11-24T15:17:00+00:00 [running]> on host 345e92ff26bf
[2024-11-24T15:18:55.635+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='fraud_team' AIRFLOW_CTX_DAG_ID='fraud_detection' AIRFLOW_CTX_TASK_ID='process_transaction' AIRFLOW_CTX_EXECUTION_DATE='2024-11-24T15:17:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-11-24T15:17:00+00:00'
[2024-11-24T15:18:55.644+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-11-24T15:18:55.754+0000] {baseoperator.py:405} WARNING - BranchPythonOperator.execute cannot be called outside TaskInstance!
[2024-11-24T15:18:55.799+0000] {fraud_detection_dag.py:78} INFO - Données récupérées depuis XCom: {"columns":["cc_num","merchant","category","amt","first","last","gender","street","city","state","zip","lat","long","city_pop","job","dob","trans_num","merch_lat","merch_long","is_fraud","current_time"],"index":[5754],"data":[[377234009633447,"fraud_Lowe, Dietrich and Erdman","kids_pets",96.54,"Theresa","Blackwell","F","43576 Kristina Islands","Shenandoah Junction","WV",25442,39.3716,-77.8229,1925,"Systems developer","1966-02-14","ee3fcf98fa6780bdcd3f9166801cfc86",39.741189,-77.235855,0,1732461528735]]}
[2024-11-24T15:18:55.801+0000] {fraud_detection_dag.py:86} INFO - Données JSON parsées avec succès
[2024-11-24T15:18:55.805+0000] {fraud_detection_dag.py:90} INFO - DataFrame créé avec 1 lignes
[2024-11-24T15:18:55.809+0000] {fraud_detection_dag.py:94} INFO - Colonnes renommées avec succès
[2024-11-24T15:18:55.809+0000] {fraud_detection_dag.py:95} INFO - Colonnes disponibles: ['cc_num', 'merchant', 'category', 'amt', 'first', 'last', 'gender', 'street', 'city', 'state', 'zip', 'lat', 'long', 'city_pop', 'job', 'dob', 'trans_num', 'merch_lat', 'merch_long', 'is_fraud', 'trans_date_trans_time']
[2024-11-24T15:18:55.812+0000] {fraud_detection_dag.py:99} INFO - Conversion de trans_date_trans_time en unix_time réussie
[2024-11-24T15:18:55.930+0000] {fraud_detection_dag.py:108} INFO - Transformation des données...
[2024-11-24T15:18:58.097+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/sklearn/ensemble/_base.py:172: UserWarning: Loky-backed parallel loops cannot be called in a multiprocessing, setting n_jobs=1
  n_jobs = min(effective_n_jobs(n_jobs), n_estimators)

[2024-11-24T15:18:58.147+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/sklearn/ensemble/_base.py:172: UserWarning: Loky-backed parallel loops cannot be called in a multiprocessing, setting n_jobs=1
  n_jobs = min(effective_n_jobs(n_jobs), n_estimators)

[2024-11-24T15:18:58.160+0000] {fraud_detection_dag.py:120} INFO - Prediction de fraude: NORMAL, Probabilité: 25.49%
[2024-11-24T15:18:58.377+0000] {fraud_detection_dag.py:126} INFO - Prédiction: NORMAL (Probabilité: 25.49%)
[2024-11-24T15:18:58.393+0000] {python.py:240} INFO - Done. Returned value was: notify_normal
[2024-11-24T15:18:58.407+0000] {branch.py:38} INFO - Branch into notify_normal
[2024-11-24T15:18:58.410+0000] {skipmixin.py:230} INFO - Following branch ('notify_normal',)
[2024-11-24T15:18:58.459+0000] {skipmixin.py:278} INFO - Skipping tasks [('skip_processing', -1), ('notify_fraud', -1)]
[2024-11-24T15:18:58.571+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-11-24T15:18:58.575+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=fraud_detection, task_id=process_transaction, run_id=scheduled__2024-11-24T15:17:00+00:00, execution_date=20241124T151700, start_date=20241124T151854, end_date=20241124T151858
[2024-11-24T15:18:58.678+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2024-11-24T15:18:58.879+0000] {taskinstance.py:3900} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2024-11-24T15:18:58.908+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
